sway-lsp/src/core/document.rs

```diff
@@ -113,39 +113,31 @@ impl TextDocument {
 /// This function ensures the necessary directory structure exists before creating the flag file.
 pub async fn mark_file_as_dirty(uri: &Url) -> Result<(), LanguageServerError> {
     let path = document::get_path_from_url(uri)?;
-    let dirty_file_path = forc_util::is_dirty_path(&path);
-    if let Some(dir) = dirty_file_path.parent() {
-        // Ensure the directory exists
-        tokio::fs::create_dir_all(dir)
-            .await
-            .map_err(|_| DirectoryError::LspLocksDirFailed)?;
-    }
-    // Create an empty "dirty" file
-    File::create(&dirty_file_path)
-        .await
-        .map_err(|err| DocumentError::UnableToCreateFile {
-            path: uri.path().to_string(),
-            err: err.to_string(),
-        })?;
-    Ok(())
+    tokio::task::spawn_blocking(move || {
```

Comment:
Why is this done in a separate thread?

Reply:
I don't think we should drop async-io as io is one of the perfect use cases for async. Instead of dropping the async I/O model, we could consider abstracting the async-to-sync conversion to make our code cleaner and more maintainable. something like...

```rust
async fn lock_lsp_path_async(path: PathBuf) -> Result<(), DirectoryError> {
    tokio::task::spawn_blocking(move || {
        PidFileLocking::lsp(path)
            .lock()
            .map_err(|_| DirectoryError::LspLocksDirFailed)
    })
    .await
    .map_err(|_| DirectoryError::LspLocksDirFailed)
}
```

Reply:
I think I misunderstood. The `mark_file_as_dirty` function can probably be written like. 

```rust
pub fn mark_file_as_dirty(uri: &Url) -> Result<(), LanguageServerError> {
    let path = document::get_path_from_url(uri)?;
    Ok(PidFileLocking::lsp(path)
        .lock()
        .map_err(|_| DirectoryError::LspLocksDirFailed)?)
}
```

Reply:
No need for this function to be async if none of the internal code needs awaiting